### DISCUSSION TITLE
Possible to move forecastTemplate to Code Behind?

### DISCUSSION THREAD

**Discussion by LorenzMichels** [PeterHimschoot/microsoft-blazor-book-3#8] [Q&A] [answered]
Hi,

Is it possible to move...
```
        private RenderFragment<WeatherForecast> forecastTemplate =
            (forecast) => @<span>@forecast.Summary</span>;
```
...from the FetchData.razor to the code behind file? 

I tried, but it generates errors.

On one hand I think that, being in the @code block in the razor file, it should work the same in code behind, but on the other hand I think @ is specific for a raxor file.

Should I use a HtmlBuilder or something like that?

Chapter 4. Advanced components

**Comment by PeterHimschoot** [comment, marked as answer]
A razor template uses razor syntax, so you cannot build it like that in the code behind.
